### ISSUE TITLE
CI error on Windows tests

### ISSUE THREAD

**Issue by zm711** [SpikeInterface/spikeinterface#3371]
## Error message
```
RuntimeError: GitHub rate limit exceeded and GITHUB_TOKEN not set; suggest setting GITHUB_TOKEN in order to get increased rate limit
```

It suggests setting a token. Should we do that to prevent this flaky test failure? @h-mayorquin (you're always the CI king). 

See this run for an example:
https://github.com/SpikeInterface/spikeinterface/actions/runs/10719779060/job/29724578046?pr=3370

**Comment by alejoe91** [comment]
I think that the issue is that at that time there were too many CI jobs in parallel. Let's wait a bit and retrigger

**Comment by zm711** [comment]
Yeah sono impaziente. Me dispiace. haha.

**Comment by h-mayorquin** [comment]
Let's keep an eye on this, we have way more runners on neuroconv and this is not a common problem there.

**Comment by zm711** [comment]
I've only started seeing it in the past month-ish and only on Windows runners. Just so you have what I have seen.

**Comment by JoeZiminski** [comment]
It might be worth setting a token to remove this as a possibility, the permitted [API calls](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28) go from 60 per hour unauthenticated to 5000 per hour with a token so should never be an issue once the token is set. I think [setting it up](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28#authenticating-in-a-github-actions-workflow) is quite simple (famous last words) but needs to be done by the repo / organisation owner.

**Comment by JoeZiminski** [comment]
Yes sounds good!

**Comment by zm711** [comment]
Have another rate limit on Windows in my PR #4033.

**Comment by zm711** [comment]
I have seen this happen 3 more times. What do you think @alejoe91 ? Are we close to doing the token?

**Comment by alejoe91** [comment]
Yep let's do that. There should be a built-in GITHUB_TOKEN so we can just set this in the workflow yaml:
```
env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
